File: Mandatory/021 - Trees and Graphs - Binary Tree/SumLeftNodes.py
import math

def leftmost_nodes_sum(array):
  row = 0
  sum = 0
  while (2 ** row < len(array)):
    sum += array[2 ** row - 1]
    row += 1
    
  return sum

print(leftmost_nodes_sum([2, 7, 5, 2, 6, 0, 9]))
# => 11

File: Mandatory/021 - Trees and Graphs - Binary Tree/SumLeftNodes.js
function leftmost_nodes_sum(array) {
  let row = 0;
  let sum = 0;
  while (Math.pow(2, row) < array.length) {
    sum += array[Math.pow(2, row) - 1]
    row += 1;
  }

  return sum;
}

console.log(leftmost_nodes_sum([2, 7, 5, 2, 6, 0, 9]));
// => 11